app/scripts/components/common/map/controls/aoi/custom-aoi-control.tsx

```diff
@@ -323,29 +323,43 @@ export default function CustomAoIControl({
   disableReason?: React.ReactNode;
 }) {
   const { main } = useMaps();
-
   const { isDrawing } = useAois();
 
   // Start/stop the drawing.
   useEffect(() => {
-    // Property was added to access draw control.
-    const mbDraw = main?._drawControl;
-    if (!mbDraw) return;
+    if (!main) return;
+
+    const changeDrawMode = () => {
+      const mbDraw = main._drawControl;
+      if (!mbDraw) return;
+
+      if (isDrawing) {
+        mbDraw.changeMode(DRAW_POLYGON);
+      } else {
+        mbDraw.changeMode(SIMPLE_SELECT, {
+          featureIds: mbDraw.getSelectedIds()
+        });
+      }
+    };
 
-    if (isDrawing) {
-      mbDraw.changeMode(DRAW_POLYGON);
+    if (main.loaded()) {
```

Comment:
Ah, I see; so the problem was that we couldn't guarantee that the map was loaded before registering mapbox-draw-related events?

Reply:
Yes, I think the issue came from two places:

1. We couldn’t guarantee the map was fully loaded before adding the Mapbox Draw events. So I added this check to make sure `main.loaded()` is `true` before switching modes when `DrawControl` is initialized
2. There was a strange synchronization issue in the previous code that initialized `useControl`. Specifically `control.current` was `undefined` later in the `CustomAoiControl` and adding safeguards or null checks didn't fix the problem. So I ended up [assigning the control instance](https://github.com/NASA-IMPACT/veda-ui/pull/1205/files#diff-d1b613ffea6e16bb9dcd389b291cc036a6e838535e87113bb3c43de86e7c2990R62) returned by `useControl` directly to a variable (like `const drawControl = useControl(...)`) to make sure we're always using the correct and initialized control instance

Reply:
oh- interesting. It will be great to figure out why 2. is happening (not a blocker, though!)